src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingGraph.java

```diff
@@ -92,278 +45,43 @@ public ReadThreadingGraph(final int kmerSize) {
      * <p>
      *     Note: only used for testing.
      * </p>
-     * @param s the string representation of the graph {@code null}.
      */
     @VisibleForTesting
     protected ReadThreadingGraph(final int kmerSizeFromString, final EdgeFactory<MultiDeBruijnVertex, MultiSampleEdge> edgeFactory) {
         super(kmerSizeFromString, new MyEdgeFactory(1));
-        debugGraphTransformations = false;
-        minBaseQualityToUseInAssembly = 0;
-    }
-
-    @VisibleForTesting
-    protected void setAlreadyBuilt() {
-        alreadyBuilt = true;
-    }
-
-    @VisibleForTesting
-    void setMaxMismatchesInDanglingHead(final int maxMismatchesInDanglingHead) {
-        this.maxMismatchesInDanglingHead = maxMismatchesInDanglingHead;
-    }
-
-    /**
-     * Return the collection of outgoing vertices that expand this vertex with a particular base.
-     *
-     * @param v original vertex.
-     * @param b expanding base.
-     * @return never null, but perhaps an empty set. You cannot assume that you can modify the result.
-     */
-    @VisibleForTesting
-    Set<MultiDeBruijnVertex> getNextVertices(final MultiDeBruijnVertex v, final byte b) {
-        Utils.nonNull(v, "the input vertex cannot be null");
-        Utils.validateArg(vertexSet().contains(v), "the vertex must be present in the graph");
-        final List<MultiDeBruijnVertex> result = new LinkedList<>();
-        for (final MultiDeBruijnVertex w : outgoingVerticesOf(v)) {
-            if (w.getSuffix() == b) {
-                result.add(w);
-            }
-        }
-        switch (result.size()) {
-            case 0: return Collections.emptySet();
-            case 1: return Collections.singleton(result.get(0));
-            default:
-                    return new HashSet<>(result);
-        }
     }
 
     /**
      * Create a new ReadThreadingAssembler using kmerSize for matching
      * @param kmerSize must be >= 1
      */
     ReadThreadingGraph(final int kmerSize, final boolean debugGraphTransformations, final byte minBaseQualityToUseInAssembly, final int numPruningSamples) {
-        super(kmerSize, new MyEdgeFactory(numPruningSamples));
-
-        Utils.validateArg( kmerSize > 0, () -> "bad minkKmerSize " + kmerSize);
-
-        this.debugGraphTransformations = debugGraphTransformations;
-        this.minBaseQualityToUseInAssembly = minBaseQualityToUseInAssembly;
-
-        resetToInitialState();
+        super(kmerSize, debugGraphTransformations, minBaseQualityToUseInAssembly, numPruningSamples);
     }
 
     /**
      * Reset this assembler to its initial state, so we can create another assembly with a different set of reads
      */
-    private void resetToInitialState() {
+    @Override
+    protected void setToInitialState() {
```

Comment:
It seems like the code here should just be inlined to the subclass constructors.  It means you probably have to change the constructor chains a bit, but I think that's ok.

Reply:
Correct. I suspect this had to do with the graph testing code at some point. I've removed this method